Platform/DataStructures/Bag.swift

```diff
@@ -143,15 +143,12 @@ struct Bag<T> : CustomDebugStringConvertible {
             return existingObject
         }
 
-        for i in 0 ..< _pairs.count {
-            if _pairs[i].key == key {
-                let value = _pairs[i].value
-                _pairs.remove(at: i)
-                return value
-            }
+
+        guard let index = _pairs.index(where: { $0.key == key }) else {
+            return nil
```

Comment:
Get the index of the pair, which needs to be removed with the build in index function instead of `for each` loop